packages/controllers/src/plugins/PluginController.test.ts

```diff
@@ -40,8 +40,61 @@ describe('PluginController Controller', () => {
         name: 'PluginController',
       }),
     });
+
     expect(pluginController).toBeDefined();
   });
+  it('can create a worker and plugin controller and add a plugin and update its state', async () => {
+    const workerExecutionEnvironment = new WebWorkerExecutionEnvironmentService(
+      {
+        setupPluginProvider: jest.fn(),
+        workerUrl: new URL(URL.createObjectURL(new Blob([workerCode]))),
+      },
+    );
+    const pluginController = new PluginController({
+      terminateAllPlugins: workerExecutionEnvironment.terminateAllPlugins.bind(
+        workerExecutionEnvironment,
+      ),
+      terminatePlugin: workerExecutionEnvironment.terminatePlugin.bind(
+        workerExecutionEnvironment,
+      ),
+      executePlugin: workerExecutionEnvironment.executePlugin.bind(
+        workerExecutionEnvironment,
+      ),
+      getRpcMessageHandler:
+        workerExecutionEnvironment.getRpcMessageHandler.bind(
+          workerExecutionEnvironment,
+        ),
+      removeAllPermissionsFor: jest.fn(),
+      getPermissions: jest.fn(),
+      hasPermission: jest.fn(),
+      requestPermissions: jest.fn(),
+      closeAllConnections: jest.fn(),
+      messenger: new ControllerMessenger<any, any>().getRestricted({
+        name: 'PluginController',
+      }),
+    });
+    const plugin = await pluginController.add({
+      name: 'TestPlugin',
+      sourceCode: `
+        wallet.registerRpcMessageHandler(async (origin, request) => {
+          const {method, params, id} = request;
+          wallet.request({method: 'setState'})
+          return method + id;
+        });
+      `,
```

Comment:
This plugin source code doesn't really make sense. I know it's only for show any way, but if we call `setState`, we should add some params to make it look realistic. Same goes for other `TestPlugin` source code instances in this file.